libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -438,15 +402,12 @@ bool AsmAnalyzer::expectDeposit(int const _deposit, int const _oldHeight, Source
 	int stackDiff = m_stackHeight - _oldHeight;
 	if (stackDiff != _deposit)
 	{
-		m_errors.push_back(make_shared<Error>(
-			Error::Type::TypeError,
-			"Expected instruction(s) to deposit " +
-			boost::lexical_cast<string>(_deposit) +
-			" item(s) to the stack, but did deposit " +
-			boost::lexical_cast<string>(stackDiff) +
-			" item(s).",
-			_location
-		));
+		m_errorReporter.typeError(
+			_location,
+			"Expected instruction(s) to deposit " + boost::lexical_cast<string>(_deposit) +
```

Comment:
Wouldn't it be cleaner to have one clear style instead if two?  Having two styles also means that, if we ever do further work on ErrorReporter, then parts of the compiler get left out.